Code/Framework/AzCore/AzCore/Date/DateFormat.h

```diff
@@ -0,0 +1,456 @@
+    template<class Duration>
+    bool GetFilenameCompatibleFormat(Iso8601TimestampString& utcTimestamp,
+        const AZStd::chrono::utc_time<Duration>& utcTime)
+    {
+        if (Iso8601TimestampString timestampExtendedDate;
+            GetIso8601ExtendedFormatDate(timestampExtendedDate, utcTime))
+        {
+            if (Iso8601TimestampString timestampBasicTime;
+                GetIso8601BasicFormatTime(timestampBasicTime, utcTime))
+            {
+                utcTimestamp = timestampExtendedDate + timestampBasicTime;
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    // formats a ISO 8601 like timestamp that is safe to use as a filename
+    // It combines the Extended ISO 8601 Date + Basic ISO 8601 Time
+    // This includes milliseconds as part of the timestamp
+    template<class Duration>
+    bool GetFilenameCompatibleFormatWithMilliseconds(Iso8601TimestampString& utcTimestamp,
+        const AZStd::chrono::utc_time<Duration>& utcTime)
+    {
+        if (Iso8601TimestampString timestampExtendedDate;
+            GetIso8601ExtendedFormatDate(timestampExtendedDate, utcTime))
+        {
+            if (Iso8601TimestampString timestampBasicTime;
+                GetIso8601BasicFormatTimeWithMilliseconds(timestampBasicTime, utcTime))
+            {
+                utcTimestamp = timestampExtendedDate + timestampBasicTime;
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    // formats a ISO 8601 like timestamp that is safe to use as a filename
+    // It combines the Extended ISO 8601 Date + Basic ISO 8601 Time
+    // This includes microseconds as part of the timestamp
+    template<class Duration>
+    bool GetFilenameCompatibleFormatWithMicroseconds(Iso8601TimestampString& utcTimestamp,
+        const AZStd::chrono::utc_time<Duration>& utcTime)
+    {
+        if (Iso8601TimestampString timestampExtendedDate;
+            GetIso8601ExtendedFormatDate(timestampExtendedDate, utcTime))
+        {
+            if (Iso8601TimestampString timestampBasicTime;
+                GetIso8601BasicFormatTimeWithMicroseconds(timestampBasicTime, utcTime))
+            {
+                utcTimestamp = timestampExtendedDate + timestampBasicTime;
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    // The GetIso8601*Now versions queries the current utc time and formats it into a fixed string
+    bool GetIso8601ExtendedFormatNow(Iso8601TimestampString& utcTimestamp);
+    bool GetIso8601ExtendedFormatNowWithMilliseconds(Iso8601TimestampString& utcTimestamp);
+    bool GetIso8601ExtendedFormatNowWithMicroseconds(Iso8601TimestampString& utcTimestamp);
+
+    bool GetIso8601BasicFormatNow(Iso8601TimestampString& utcTimestamp);
+    bool GetIso8601BasicFormatNowWithMilliseconds(Iso8601TimestampString& utcTimestamp);
+    bool GetIso8601BasicFormatNowWithMicroseconds(Iso8601TimestampString& utcTimestamp);
```

Comment:
suggestion: Might it be useful to include a sample string for each of these functions similar to what you've done in the PR description to make it clear to the reader what output they'd expect to get? Could be helpful to know at a glance what the output would be.

Reply:
Or maybe better, a path to the tests, as these are guaranteed to be correct and could be useful for people to check to see the format in the tests themselves?

Reply:
We could link to the ISO 8601 wikipedia page which shows the expected timestamp formats: https://en.wikipedia.org/wiki/ISO_8601#Calendar_dates

Reply:
Yeah that's a good compromise 👍